commerce/domain/checkout/giftcard/GiftCardAdapterSample.cls

```diff
@@ -0,0 +1,107 @@
+/**
+ * @description Custom Gift card adapter. This class will have operations like applyGiftCard(), redeemGiftCard(), refundGiftCard().
+ In this sample implementation of applyGiftCard, if the https service is registered , we make a call to that third party service
+ with giftCardNumber and pin passed as the input in the request. The response is expected to have availableAmount (in the gift card)
+ and giftCardReference. If https service is not registered then we return a hardcoded values.
+ */
+
+global class GiftCardAdapter extends CartExtension.GiftCardAdapter{
+
+    private static String httpHost = 'https://giftCardProvider.com';
+    private static Boolean useHTTPService = false;
+
+    public virtual override CartExtension.ApplyGiftCardResponse applyGiftCard(CartExtension.ApplyGiftCardRequest applyRequest){
+
+        Double amount = applyRequest.getAmount();
+        String giftCardCurrency = applyRequest.getCurrencyIsoCode();
+        String giftCardCode = applyRequest.getGiftCardCode();
+        String pin = applyRequest.getPin();
+
+        CartExtension.ApplyGiftCardResponse response = null;
+
+        if(useHTTPService) {
+           response = applyGiftCardService(giftCardCode, pin, amount);
+        } else {
+           response = applyGiftCardMockedService(giftCardCode, pin, amount);
+        }
+
+        return response;
+
+    }
+
+    private CartExtension.ApplyGiftCardResponse applyGiftCardMockedService(String giftCardCode, String giftCardPin, Double amount) {
+        CartExtension.ApplyGiftCardResponse applyResponse = new CartExtension.ApplyGiftCardResponse();
+
+        applyResponse.setGiftCardReference('123456789');
+        applyResponse.setAppliedAmount(50);
+        applyResponse.setLastFour('6789');
+        applyResponse.setCreditType('GiftCard');
```

Comment:
This should be the ENUM value from WebCartCredit entity.

Reply:
Shouldn't this value be set our javaService, why do we want this to be updated by Apex?